File: leetcode2418/solution.cpp
/*
Solution 1:

Time: O(n log n) | Space: O(n)

- n: length of names
*/

class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        map<int, string> m;
        for (int i = 0; i < names.size(); ++i) {
            m.emplace(heights[i], names[i]);
        }

        vector<string> res;

        for (auto i = m.rbegin(); i != m.rend(); ++i) {
            res.push_back(i->second);
        }

        return res;
    }
};

File: leetcode2418/solution.rs
/*
Solution 1:

Time: O(nlog(n)) | Space: O(n)

- n: length of names
 */

impl Solution {
    pub fn sort_people(names: Vec<String>, heights: Vec<i32>) -> Vec<String> {
        let mut people = names.into_iter().zip(heights.into_iter()).collect::<Vec<(String, i32)>>();

        people.sort_unstable_by_key(|p| -p.1);

        people.into_iter().map(|p| p.0).collect::<Vec<String>>()
    }
}